soc/nordic/ironside/include/nrf_ironside/uicr_def.h

```diff
@@ -0,0 +1,218 @@
+	__IOM uint32_t ADDRESS;  /* Start address of the array of peripheral configuration entries*/
+	__IOM uint32_t MAXCOUNT; /* Maximum number of peripheral configuration entries */
+};
+
+/**
+ * @brief MPCCONF [UICR_MPCCONF] Global domain MPC configuration
+ */
+struct UICR_MPCCONF {
+	__IOM uint32_t ENABLE;   /* Enable the global domain MPC configuration */
+	__IOM uint32_t ADDRESS;  /* Start address of the array of MPC configuration entries*/
+	__IOM uint32_t MAXCOUNT; /* Maximum number of MPC configuration entries */
+};
+
+/**
+ * @brief TRIGGER [UICR_SECONDARY_TRIGGER] Automatic triggers for reset into secondary firmware
+ */
+struct UICR_SECONDARY_TRIGGER {
+	__IOM uint32_t ENABLE; /* Enable automatic triggers for reset into secondary firmware*/
+
+	__IOM uint32_t
+		RESETREAS; /* Reset reasons that trigger automatic reset into secondary firmware*/
+	__IM uint32_t RESERVED;
+};
+
+/* SECONDARY TRIGGER RESETREAS field access macros */
+
+#define UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONWDT0_Pos (0UL)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONWDT0_Msk                                       \
+	(0x1UL << UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONWDT0_Pos)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONWDT1_Pos (1UL)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONWDT1_Msk                                       \
+	(0x1UL << UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONWDT1_Pos)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONLOCKUP_Pos (3UL)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONLOCKUP_Msk                                     \
+	(0x1UL << UICR_SECONDARY_TRIGGER_RESETREAS_APPLICATIONLOCKUP_Pos)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCOREWDT0_Pos (5UL)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCOREWDT0_Msk                                         \
+	(0x1UL << UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCOREWDT0_Pos)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCOREWDT1_Pos (6UL)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCOREWDT1_Msk                                         \
+	(0x1UL << UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCOREWDT1_Pos)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCORELOCKUP_Pos (8UL)
+#define UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCORELOCKUP_Msk                                       \
+	(0x1UL << UICR_SECONDARY_TRIGGER_RESETREAS_RADIOCORELOCKUP_Pos)
+
+/**
+ * @brief SECONDARY [UICR_SECONDARY] Secondary firmware configuration
+ */
+struct UICR_SECONDARY {
+	__IOM uint32_t ENABLE;    /* Enable booting of secondary firmware. */
+	__IOM uint32_t PROCESSOR; /* Processor to boot for the secondary firmware. */
+
+	__IOM struct UICR_SECONDARY_TRIGGER
+		TRIGGER;        /* Automatic triggers for reset into secondary firmware */
+	__IOM uint32_t ADDRESS; /* Start address of the secondary firmware. This value is used as
+				 * the initial value of the secure VTOR (Vector Table Offset
+				 * Register) after CPU reset.
+				 */
```

Comment:
I think the code would format more nicely if we moved these comments above the variable instead of having them at the end of the line

Reply:
fixed